crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -225,6 +228,45 @@ impl TimelineState {
         txn.commit();
     }
 
+    /// Try to fetch [`EncryptionInfo`] for the events with the supplied
+    /// indices, and update them where we succeed.
+    pub(super) async fn retry_event_encryption_info<P: RoomDataProvider>(
```

Comment:
I have to say that having this here but `retry_event_decryption_by_index` in the other place feels rather asymmetric. Is there a reason they need to be different?

Reply:
In my mind, `TimelineState::retry_event_encryption_info` is a partner to `TimelineState::retry_event_decryption`, not to `TimelineController::retry_event_decryption_by_index`. `retry_event_decryption_by_index` is needed because we need to construct a complex `retry_one` closure to pass to `retry_event_decryption` whereas `retry_event_encryption_info` can do its work directly.